packages/react-router-dev/vite/plugin.ts

```diff
@@ -804,9 +815,14 @@ export const reactRouterVitePlugin: ReactRouterVitePlugin = () => {
           ssr: {
             external: ssrExternals,
             resolve: {
-              conditions: viteCommand === "build" ? [] : ["development"],
+              conditions:
+                viteCommand === "build"
+                  ? viteServerConditions
+                  : ["development", ...viteServerConditions],
               externalConditions:
-                viteCommand === "build" ? [] : ["development"],
+                viteCommand === "build"
+                  ? viteServerConditions
+                  : ["development", ...viteServerConditions],
```

Comment:
I would add a comment and explain why we need this, and link to: https://vite.dev/guide/migration#default-value-for-resolve-conditions

Something like:

```
// Vite 6 no longer adds default conditions internally.
// We need to explicitly include them to ensure proper module resolution
// especially for packages that rely on specific conditions like @prisma/client
// see {link}
```

Maybe it's just me, but this code is quite critical as it handles module resolution, which is a critical part of how dependencies get loaded in the application

i would prefer being very clear here

Reply:
Good idea, I added some comments.